server/src/main/java/org/elasticsearch/action/support/master/TransportMasterNodeAction.java

```diff
@@ -82,6 +84,15 @@ protected TransportMasterNodeAction(String actionName, boolean canTripCircuitBre
     protected abstract void masterOperation(Task task, Request request, ClusterState state,
                                             ActionListener<Response> listener) throws Exception;
 
+    private void executeMasterOperation(Task task, Request request, ClusterState state,
+                                        ActionListener<Response> listener) throws Exception {
+        if (task instanceof CancellableTask && ((CancellableTask) task).isCancelled()) {
+            throw new CancellationException("Task was cancelled");
```

Comment:
I don't think we exercise this in the tests any more? At least I removed it and the tests still passed several hundred iterations.

Reply:
🤦 I've added a new test for this scenario in https://github.com/elastic/elasticsearch/pull/72157/commits/c1da23f479887433991192a1de2d5244b2c2c602